### PR TITLE
chore(CI) :Nightly dev 23.10.x

### PR DESCRIPTION
**Pull request by MedWassCh**
## Description
This PR is dedicated for having a nightly workflow for the dev 23.10.x

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [ ] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/web.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 name: web
+run-name:
+  ${{ github.event_name == 'schedule' && format('web nightly {0}', github.ref ) ||
+  github.event_name == 'pull_request' && github.event.pull_request.title ||
+  github.event_name == 'push' && github.event.head_commit.message ||
+  github.event_name == 'workflow_dispatch' && github.workflow || 'unknown' }}
```
Comment (sc979) [inline .github/workflows/web.yml:6]:
```suggestion
  ${{ (github.event_name == 'schedule' && format('web nightly {0}', github.ref )) ||
  (github.event_name == 'pull_request' && github.event.pull_request.title) ||
  (github.event_name == 'push' && github.event.head_commit.message) ||
  (github.event_name == 'workflow_dispatch' && github.workflow) || 'unknown' }}
```

File: .github/workflows/web.yml [hunk 2]
```diff
@@ -83,6 +89,25 @@ jobs:
         with:
           features_path: centreon/tests/e2e/features
 
+  dispatch-to-maintained-branches:
+    if: ${{ github.event_name == 'schedule' && github.ref_name == 'develop' }}  ||
+      github.event_name == 'pull_request' && github.event.pull_request.title ||
+      github.event_name == 'push' && github.event.head_commit.message ||
+      github.event_name == 'workflow_dispatch' && github.workflow || 'unknown' }}
```
Comment (sc979) [inline .github/workflows/web.yml:96]:
```suggestion
    if: ${{ (github.event_name == 'schedule' && github.ref_name == 'develop')  ||
      (github.event_name == 'pull_request' && github.event.pull_request.title) ||
      (github.event_name == 'push' && github.event.head_commit.message) ||
      (github.event_name == 'workflow_dispatch' && github.workflow) || 'unknown' }}
```

Comment (sc979) [reply .github/workflows/web.yml:96]: what is "unknown" for ?

File: .github/workflows/web.yml (hunk 1 above)
Comment (sc979) [reply .github/workflows/web.yml:6]: what is "unknown" for ?

File: .github/workflows/web.yml [hunk 3]
```diff
@@ -83,6 +89,25 @@ jobs:
...
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - run: |
+          gh workflow run web.yml -r "dev-23.10.x"
+          gh workflow run web.yml -r "dev-23.04.x"
+          gh workflow run web.yml -r "dev-22.10.x"
+          gh workflow run web.yml -r "dev-22.04.x"
```
Comment (kduret) [inline .github/workflows/web.yml:106]: you could add an boolean input to the workflow which defined if it is a nightly build
this should be false by default
(environment input may be used also)
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-of-onworkflow_dispatchinputs
```suggestion
          gh workflow run web.yml -r "dev-23.10.x" -f is_nightly=true
          gh workflow run web.yml -r "dev-23.04.x" -f is_nightly=true
          gh workflow run web.yml -r "dev-22.10.x" -f is_nightly=true
          gh workflow run web.yml -r "dev-22.04.x" -f is_nightly=true
```

**Review by kduret — COMMENTED** [review]
those changes should target develop branch

**Pull request closed without merging**